File: lg/src/questions/q_0200_number_of_islands_test.go
package questions

import (
	"testing"

	"github.com/stretchr/testify/require"
)

func TestNumIslands(t *testing.T) {
	tests := []struct {
		grid   [][]byte
		expect int
	}{
		{
			grid: [][]byte{
				[]byte("11110"),
				[]byte("11010"),
				[]byte("11000"),
				[]byte("00000"),
			},
			expect: 1,
		},
		{
			grid: [][]byte{
				[]byte("11000"),
				[]byte("11000"),
				[]byte("00100"),
				[]byte("00011"),
			},
			expect: 3,
		},
	}

	for _, test := range tests {
		require.Equal(t, test.expect, numIslands(test.grid))
	}
}

File: lt/src/q0200NumberOfIslands.test.ts
import { numIslands } from "./q0200NumberOfIslands";

test("q0200NumberOfIslands", () => {
    const grid = [
        ["1", "1", "1", "1", "0"],
        ["1", "1", "0", "1", "0"],
        ["1", "1", "0", "0", "0"],
        ["0", "0", "0", "0", "0"],
    ];
    expect(numIslands(grid)).toBe(1);
    const grid2 = [
        ["1", "1", "0", "0", "0"],
        ["1", "1", "0", "0", "0"],
        ["0", "0", "1", "0", "0"],
        ["0", "0", "0", "1", "1"],
    ];
    expect(numIslands(grid2)).toBe(3);
});
